src/buffer/buffer_map.rs

```diff
@@ -373,33 +382,99 @@ mod tests {
             let buffer_f64 = builder.create_buffer(BufferSettings::default());
             let buffer_string = builder.create_buffer(BufferSettings::default());
             let buffer_generic = builder.create_buffer(BufferSettings::default());
+            let buffer_any = builder.create_buffer::<()>(BufferSettings::default());
+
+            scope.input.chain(builder).fork_unzip((
+                |chain: Chain<_>| chain.connect(buffer_i64.input_slot()),
+                |chain: Chain<_>| chain.connect(buffer_f64.input_slot()),
+                |chain: Chain<_>| chain.connect(buffer_string.input_slot()),
+                |chain: Chain<_>| chain.connect(buffer_generic.input_slot()),
+                |chain: Chain<_>| chain.connect(buffer_any.input_slot()),
+            ));
 
             let buffers = TestJoinedValue::select_buffers(
-                builder,
                 buffer_i64,
                 buffer_f64,
                 buffer_string,
                 buffer_generic,
+                buffer_any.into(),
+            );
+
+            builder.join(buffers).connect(scope.terminate);
+        });
+
+        let mut promise = context.command(|commands| {
+            commands
+                .request(
+                    (5_i64, 3.14_f64, "hello".to_string(), "world", ()),
+                    workflow,
+                )
+                .take_response()
+        });
+
+        context.run_with_conditions(&mut promise, Duration::from_secs(2));
+        let value: TestJoinedValue<&'static str> = promise.take().available().unwrap();
+        assert_eq!(value.integer, 5);
+        assert_eq!(value.float, 3.14);
+        assert_eq!(value.string, "hello");
+        assert_eq!(value.generic, "world");
+        assert!(context.no_unhandled_errors());
+    }
+
+    #[test]
+    fn test_select_buffers_json() {
+        let mut context = TestingContext::minimal_plugins();
+
+        let workflow = context.spawn_io_workflow(|scope, builder| {
+            let buffer_i64 =
+                JsonBuffer::from(builder.create_buffer::<i64>(BufferSettings::default()));
+            let buffer_f64 =
+                JsonBuffer::from(builder.create_buffer::<f64>(BufferSettings::default()));
+            let buffer_string =
+                JsonBuffer::from(builder.create_buffer::<String>(BufferSettings::default()));
+            let buffer_generic =
+                JsonBuffer::from(builder.create_buffer::<String>(BufferSettings::default()));
+            let buffer_any =
+                JsonBuffer::from(builder.create_buffer::<()>(BufferSettings::default()));
```

Comment:
You may want to move this test into the `json_buffer.rs` module since `JsonBuffer` requires the `diagram` feature, but `buffer_map.rs` doesn't.